tools/test/CMakeLists.txt

```
@@ -13,15 +13,26 @@ add_dependencies(ret0 not)
 llvm_test_run(EXECUTABLE "$<TARGET_FILE:not>" "$<TARGET_FILE:not>" "$<TARGET_FILE:ret0>")
 llvm_add_test_for_target(ret0)
 
-# Check that expected crashes are handled correctly.
-llvm_test_executable_no_test(abrt abort.c)
-add_dependencies(abrt not)
-llvm_test_run(EXECUTABLE "$<TARGET_FILE:not>" "--crash" "$<TARGET_FILE:abrt>")
-llvm_add_test_for_target(abrt)
+# These test will always fail under user-mode emulation because 'not'
+# spawns a subprocess outside the emulator and the check_env test
+# runs the host python interpreter under the emulator for the target.
+if(NOT(TEST_SUITE_USER_MODE_EMULATION AND TEST_SUITE_RUN_UNDER))
```

The negation of the conditional is a bit hard to read. Could the branches be switched with a "positive" conditional instead?

```suggestion
if(TEST_SUITE_USER_MODE_EMULATION AND TEST_SUITE_RUN_UNDER)
```